src/views/Profile/PersonalInfo/PersonalInfo.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { Avatar, Grid, TextField, InputLabel } from '@material-ui/core';
+import { useStyles } from './styles/PersonalInfo.styles';
+import { IExpert } from '../../../old/lib/types';
+import { getCurrentUser } from '../../../old/lib/utilities/API/api';
+import { ERROR_404 } from '../../../old/lib/constants/routes';
+
+export const PersonalInfo: React.FC = () => {
+  const classes = useStyles();
+  const { expertId } = useParams<{ expertId: string }>();
+  const [loadedExpert, setLoadedExpert] = useState<IExpert>();
+  const [statusCode, setStatusCode] = useState<number>();
+  const history = useHistory();
+
+  const fetchExpert = useCallback(async () => {
+    try {
+      const expertResponse = await getCurrentUser();
+      setLoadedExpert(expertResponse.data);
+    } catch (error) {
+      setStatusCode(404);
```

Comment:
I think, redirecting from this place right to `ERROR_404` router is more clear that using additional variable to redirect on next re-render. Agree?